helper/schema/field_writer_map.go

```diff
@@ -277,7 +277,7 @@ func (w *MapFieldWriter) setSet(
 		// not the `value` directly is because this forces all types
 		// to become []interface{} (generic) instead of []string, which
 		// most hash functions are expecting.
-		s := &Set{F: schema.Set}
```

Comment:
Oh my... this must've been lurking here since I did the refactoring that added `schema.ZeroValue` months ago. Sorry about that!

Curious to see if this will fix weird quirky behavior with *other* resources that use the default hash implementation. I know a few times I've seen "diffs don't match" errors that I then failed to reproduce, and it seems plausible that this would be the cause for at least some of them...